### DISCUSSION TITLE
[Auth] Display user valid sessions and allow to invalidate some or all of them

### DISCUSSION THREAD

**Discussion by tomekit** [supabase/supabase#26863] [Feature Requests]
Is there a way to display user active refresh tokens (e.g. with some associated information / device type) where user is able to display  their valid sessions and invalidate some of them without affecting others?

By saying session I mean refresh token, not access token. I am aware of that access token can't be revoked, it will expire itself in a sensible time.

As a user think of a scenario when you're selling phone and restored device to factory settings, but forgot to "logout". In such case valid refresh token stays in the Supabase Auth, it would be good to be aware of it  and invalidate it whilst preserving session on other devices.

It's not uncommon for other auth solutions to have an option to build a page for a user page with valid sessions, device type, last used and e.g. IP address.

I think it would be beneficial for Supabase to allow similar level of session granularity.

Related: https://github.com/orgs/supabase/discussions/26863
[Auth] Allow indefinite refresh token reuse

**Comment by guillempuche** [comment]
Good one!

Great secure platforms (Facebook, Google...), as a security feature, allow their users to close active sessions via remotely.

It would be great to allow developers to have a list of devices per user (browser session, mobile app...) with privacy in mind (optional IP, optional fingerprint...), last date logged in, and customizable device name.

**Comment by guillempuche** [comment]
Also requested in https://github.com/orgs/supabase/discussions/26826
